src/pylint_odoo/checkers/import_checker.py

```diff
@@ -0,0 +1,128 @@
+import re
+from typing import Union
+
+from astroid import Import, ImportFrom, Module
+from astroid.modutils import modpath_from_file
+from pylint.checkers.utils import only_required_for_messages
+from pylint.lint import PyLinter
+
+from .odoo_base_checker import OdooBaseChecker
+
+ODOO_MSGS = {
+    # C->convention R->refactor W->warning E->error F->fatal
+    "E8401": (
+        "This python module is not imported and has no effect",
+        "unused-module",
+        "",
+    ),
+}
+
+EXCLUDED = re.compile("__manifest__.py$|__openerp__.py$|__init__.py$|tests/|migrations/|tests\\\\|migrations\\\\")
+
+
+class ImportChecker(OdooBaseChecker):
+    msgs = ODOO_MSGS
+    name = "odoolint"
+
+    def __init__(self, linter: PyLinter):
+        self.imports = set()
+        self.modules = set()
+
+        super().__init__(linter)
+
+    @staticmethod
+    def get_module_from_node(node, max_attempts: int = 15) -> Union[None, Module]:
+        """Get the module a node belongs to.
+
+        :param node: Node that belongs to the module to be extracted.
+        :param int max_attempts: Amount of attempts that will be made to obtain the module. Nodes that are
+        nested deeper than max_attempts won't be found.
+
+        :returns: Module if found, otherwise None.
+        """
+        attempts = 0
+        while attempts < max_attempts:
```

Comment:
This may be simplified a bit:

`for attempt in range(max_attempts)`

That way, you won't need to take care of increasing manually the number of attempts